ioservice/io_fops.c

```diff
@@ -1829,97 +1370,6 @@ static void io_item_replied(struct m0_rpc_item *item)
 	reply = io_rw_rep_get(rfop);
 	M0_ASSERT(ergo(reply->rwr_rc == 0,
 		       reply->rwr_count == rbulk->rb_bytes));
-
-if (0) /* if (0) is used instead of #if 0 to avoid code rot. */
-{
-	/** @todo Rearrange IO item merging code to work with new
-		  formation code.
-	 */
-	/*
-	 * Restores the contents of main coalesced fop from the first
-	 * rpc item in m0_rpc_item::ri_compound_items list. This item
-	 * is inserted by io coalescing code.
-	 */
-	if (!rpcitem_tlist_is_empty(&item->ri_compound_items)) {
-		M0_LOG(M0_DEBUG, "Reply received for coalesced io fops.");
-		ritem = rpcitem_tlist_pop(&item->ri_compound_items);
-		bkpfop = m0_rpc_item_to_fop(ritem);
-		if (fop->f_type->ft_ops->fto_fop_replied != NULL)
-			fop->f_type->ft_ops->fto_fop_replied(fop, bkpfop);
-	}
-
-	/*
-	 * The rpc_item->ri_chan is signaled by sessions code
-	 * (rpc_item_replied()) which is why only member coalesced items
-	 * (items which were member of a parent coalesced item) are
-	 * signaled from here as they are not sent on wire but hang off
-	 * a list from parent coalesced item.
-	 */
-	m0_tl_for(rpcitem, &item->ri_compound_items, ritem) {
-		fop = m0_rpc_item_to_fop(ritem);
-		rbulk = m0_fop_to_rpcbulk(fop);
-		m0_mutex_lock(&rbulk->rb_mutex);
-		M0_ASSERT(rbulk != NULL && m0_tlist_is_empty(&rpcbulk_tl,
-			  &rbulk->rb_buflist));
-		/* Notifies all member coalesced items of completion status. */
-		rbulk->rb_rc = item->ri_error;
-		m0_mutex_unlock(&rbulk->rb_mutex);
-		/* XXX Use rpc_item_replied()
-		       But we'll fix it later because this code path will need
-		       significant changes because of new formation code.
-		 */
-		/* m0_chan_broadcast(&ritem->ri_chan); */
-	} m0_tl_endfor;
-}
-}
-
-static void item_io_coalesce(struct m0_rpc_item *head, struct m0_list *list,
-			     uint64_t size)
-{
-	/* Coalescing RPC items is not yet supported */
-if (0)
-{
-	int			 rc;
-	struct m0_fop		*bfop;
-	struct m0_rpc_item	*item;
-
-	M0_PRE(head != NULL);
-	M0_PRE(list != NULL);
-	M0_PRE(size > 0);
-
-	if (m0_list_is_empty(list))
-		return;
-
-	/*
-	 * Traverses through the list and finds out items that match with
-	 * head on basis of fid and intent (read/write). Matching items
-	 * are removed from session->s_unbound_items list and added to
-	 * head->compound_items list.
-	 */
-	bfop = m0_rpc_item_to_fop(head);
-
-	if (rpcitem_tlist_is_empty(&head->ri_compound_items))
-		return;
-
-	/*
-	 * Add the bound item to list of compound items as this will
-	 * include the bound item's io vector in io coalescing
-	 */
-	rpcitem_tlist_add(&head->ri_compound_items, head);
```

Comment:
Can we get rid of `m0_rpc_item::ri_compound_items` as well? It looks like they are used only in this "coalescing" code.